RecoverCRPVM/Common-Functions.psm1

```diff
@@ -0,0 +1,408 @@
+        }
+
+
+        <#if ($ostype -eq "Linux")
+        {
+            $sshPublicKey = Get-Content "$env:USERPROFILE\.ssh\id_rsa.pub"
+            $rescuevm = Add-AzureRmVMSshPublicKey -VM $rescuevm -KeyData $sshPublicKey -Path "/home/azureuser/.ssh/authorized_keys"
+        }#>
+
+
+        ## Create the VM in Azure
+        Write-Log "Creating new Resuce VM name ==> $($rescuevm.Name) under ResourceGroup ==> $RescueResourceGroup" -Color Yellow
+        $created = New-AzureRmVM -ResourceGroupName $RescueResourceGroup -Location $Location -VM $rescuevm 
+        Write-Log "Successfully created Rescue VM ==> $rescueVMNname was created under ResourceGroup==> $RescueResourceGroup" -Color Green 
+       
+        Return $created
+    }
+    Catch
+    {
+        Write-Log "Unable to create the rescue VM successfully" -Color Red
+        Write-Log "Unable to create the rescue VM successfully - -  Exception Type: $($_.Exception.GetType().FullName)" -logOnly
+        Write-Log "Exception Message: $($_.Exception.Message)" -logOnly
+        throw
+        return null
+    }  
+    
+}
+
+function AttachOsDisktoRescueVM(
+[String]$RescueResourceGroup,
+[String]$rescueVMNname,
+[String]$osDiskVHDToBeRepaired,
+[String]$VHDNameShort,
+[String]$osDiskSize,
+[String]$managedDiskID
+)
+{
+    $returnVal = $true
+    Write-Log "Running Get-AzureRmVM -ResourceGroupName `"$RescueResourceGroup`" -Name `"rescueVMNname`"" -Color Yellow
+    $rescuevm = Get-AzureRmVM -ResourceGroupName $RescueResourceGroup -Name $rescueVMNname
+    if (-not $rescuevm)
+    {
+        Write-Log "RescueVM ==>  $rescueVMNname cannot be found, Cannot proceed" -Color Red
+        $returnVal = $false
+    }
+    Write-Log "Attaching the OS Disk to the rescueVM" -Color Yellow
+    Try
+    {
+        if ($managedDiskID) 
+        {
+           Add-AzureRmVMDataDisk -VM $rescueVm -Name $VHDNameShort -CreateOption Attach -ManagedDiskId $managedDiskID -Lun 0
+        }
+        else
+        {
+          Add-AzureRmVMDataDisk -VM $rescueVm -Name $VHDNameShort -Caching None -CreateOption Attach -DiskSizeInGB $osDiskSize -Lun 0 -VhdUri $osDiskVHDToBeRepaired
+        }
+        Update-AzureRmVM -ResourceGroupName $RescueResourceGroup -VM $rescuevm 
+        Write-Log "Successfully attached the OS Disk as a Data Disk" -Color Green
+    }
+    Catch
+    {
+         $returnVal = $false
+         Write-Log "Unable to Attach the OSDisk -  Exception Type: $($_.Exception.GetType().FullName)" -logOnly
+         Write-Log "Exception Message: $($_.Exception.Message)" -logOnly
+         throw
+         return $returnVal
+    }
+    return $returnVal
+}
+
+
+function StopTargetVM(
+    [String]$ResourceGroup,
+    [String]$VmName
+)
+{
+
+    Write-Log "Stopping Azure VM $VmName" -Color Yellow
+    $stopped = Stop-AzureRmVM -ResourceGroupName $ResourceGroup -Name $VmName -Force
```

Comment:
For each and every command that we are issuing, if it's failing, it is better to log the last error as well. That will help us to debug further., This is a generic commend for all the operations.